pkg/model/gguf_parser_wrapper_test.go

```diff
@@ -0,0 +1,218 @@
+
+	"github.com/golang/mock/gomock"
+	ggufparser "github.com/gpustack/gguf-parser-go"
+	"github.com/mudler/LocalAI/pkg/model"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+// MockGGUFParser is used to mock the gguf-parser methods
+type MockGGUFParser struct {
+	mock.Mock
+}
+
+type MockModelMemoryEstimator struct {
+	mock.Mock
+}
+
+func (m *MockModelMemoryEstimator) Estimate(ggufFile *ggufparser.GGUFFile) (*model.ModelEstimate, error) {
+	args := m.Called(ggufFile)
+	return args.Get(0).(*model.ModelEstimate), args.Error(1)
+}
+
+func (m *MockGGUFParser) ParseGGUFFileRemote(ctx context.Context, url string) (*ggufparser.GGUFFile, error) {
+	args := m.Called(ctx, url)
+	return args.Get(0).(*ggufparser.GGUFFile), args.Error(1)
+}
+
+func (m *MockGGUFParser) ParseGGUFFileFromOllama(ctx context.Context, model string) (*ggufparser.GGUFFile, error) {
+	args := m.Called(ctx, model)
+	return args.Get(0).(*ggufparser.GGUFFile), args.Error(1)
+}
+
+func (m *MockGGUFParser) ParseGGUFFileFromHuggingFace(ctx context.Context, repo, file string) (*ggufparser.GGUFFile, error) {
+	args := m.Called(ctx, repo, file)
+	return args.Get(0).(*ggufparser.GGUFFile), args.Error(1)
+}
+
+func (m *MockGGUFParser) ParseGGUFFile(filePath string) (*ggufparser.GGUFFile, error) {
+	args := m.Called(filePath)
+	return args.Get(0).(*ggufparser.GGUFFile), args.Error(1)
+}
+
+// MockEstimateModelMemoryUsage mocks the internal function estimateModelMemoryUsage for testing
+func MockEstimateModelMemoryUsage(mockGGUFFile *ggufparser.GGUFFile) *model.ModelEstimate {
+	return &model.ModelEstimate{
+		Estimate: model.ModelEstimateItems{
+			Items: []model.ModelMemory{
+				{
+					OffloadLayers: 32,
+					FullOffloaded: true,
+					RAM: model.EstimateRAM{
+						UMA:    512,
+						NonUMA: 1024,
+					},
+					VRAMs: []model.EstimateVRAM{
+						{
+							UMA:    2048,
+							NonUMA: 4096,
+						},
+					},
+				},
+			},
+			Type:              "model",
+			Architecture:      "llama",
+			ContextSize:       2048,
+			FlashAttention:    true,
+			NoMMap:            false,
+			EmbeddingOnly:     false,
+			Distributable:     true,
+			LogicalBatchSize:  2048,
+			PhysicalBatchSize: 512,
+		},
+		Architecture: model.Architecture{
+			Type:                 "model",
+			Architecture:         "llama",
+			MaximumContextLength: 2048,
+			EmbeddingLength:      512,
+			VocabularyLength:     32000,
+		},
+		Metadata: model.Metadata{
+			Type:                "model",
+			Architecture:        "llama",
+			QuantizationVersion: 2,
+			Alignment:           32,
+			Name:                "Mock Model",
+			License:             "open",
+			FileType:            1,
+			LittleEndian:        true,
+			FileSize:            1024,
+			Parameters:          1000000,
+		},
+		Tokenizer: model.Tokenizer{
+			Model:        "gpt2",
+			TokensLength: 32000,
+			TokensSize:   512000,
+		},
+	}
+}
+
+func TestGetModelGGufData_URL_WithMockedEstimateModelMemoryUsage(t *testing.T) {
```

Comment:
This is still a minor nit, but all other tests are using ginkgo - do you feel to use ginkgo as well? not a blocker in any case, at some point will refactor things out to be more consistent if needed